lib/fog/vsphere/requests/compute/create_vm.rb

```diff
@@ -122,18 +122,23 @@ def controller_get_shared_from_options options
         end
 
         def create_disk disk, index = 0, operation = :add, controller_key = 1000
+          if (index > 6) then
+            _index = index + 1
+          else
+            _index = index
+          end
```

Comment:
@plribeiro3000  - Only incrementing the index for 7 will resulting in losing a drive (if you have ore than 7 drives)
What about the actual drive that has the index of 8, this will be ignored by only incrementing if index == 7, because the drive with initial index of 7 will now be the drive with index of 8.
Basically we need to create a 'hole' in the array at index 7.